### PR TITLE
Add support for Ed25519 certificates

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation

While Ed25519 is not available in the WebPKI, there are a number of contexts where it's an appropriate signature algorithm for X.509. In those contexts,
we should allow the signature algorithm.

Modifications

Add support for Ed25519 keys and signatures.
Add unit tests

Result

Ed25519 support in the X509 certs

### CODE REVIEW

**Review by FranzBusch — APPROVED** [review]

File: Sources/_CertificateInternals/CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@
 ##===----------------------------------------------------------------------===##
 
 add_library(_CertificateInternals
-  "_TinyArray.swift")
+)
```
Comment (3405691582) [inline Sources/_CertificateInternals/CMakeLists.txt:16]: Is this change intended? I get a cmake error now this is a library without any sources.

Comment (Lukasa) [reply Sources/_CertificateInternals/CMakeLists.txt:16]: Hrm, that looks incorrect but CI should have caught it.
